src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs

```diff
@@ -0,0 +1,312 @@
+
+            private static B? GetB(bool create) => create ? new B() : default(B);
+
+            private static A? GetAB(string? text) => new A { B = new B { Text = text } };
+                
+            private class A
+            {
+                public B? B { get; set; }
+            }
+
+            private class B
+            {
+                public string? Text { get; set; }
+
+                public void Clear() => this.Text = null;
+
+                [System.Diagnostics.Contracts.Pure]
+                public string SafeGetText() => this.Text ?? string.Empty;
+            }
+        ";
+
+        private static readonly DiagnosticSuppressor suppressor = new DereferencePossiblyNullReferenceSuppressor();
+
+        [TestCase("")]
+        [TestCase("Assert.NotNull(string.Empty)")]
+        [TestCase("Assert.IsNull(s)")]
+        [TestCase("Assert.Null(s)")]
+        [TestCase("Assert.That(s, Is.Null)")]
+        public async Task NoValidAssert(string assert)
+        {
+            var testCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@$"
+                #nullable enable
+                [TestCase("""")]
+                public void Test(string? s)
+                {{
+                    {assert};
+                    Assert.That(s.Length, Is.GreaterThan(0));
+                }}
+            ");
+
+            await DiagnosticsSuppressorAnalyzer.EnsureNotSuppressed(suppressor, testCode)
+                                               .ConfigureAwait(false);
+        }
+
+        [TestCase("Assert.NotNull(s)")]
+        [TestCase("Assert.IsNotNull(s)")]
+        [TestCase("Assert.That(s, Is.Not.Null)")]
+        public async Task WithLocalValidAssert(string assert)
+        {
+            var testCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@$"
+                #nullable enable
+                [TestCase("""")]
+                public void Test(string? s)
+                {{
+                    {assert};
+                    Assert.That(s.Length, Is.GreaterThan(0));
+                }}
+            ");
+
+            await DiagnosticsSuppressorAnalyzer.EnsureSuppressed(suppressor,
+                DereferencePossiblyNullReferenceSuppressor.SuppressionDescriptors["CS8602"], testCode)
+                .ConfigureAwait(false);
+        }
+
+        [TestCase("Assert.NotNull(s)")]
+        [TestCase("Assert.IsNotNull(s)")]
+        [TestCase("Assert.That(s, Is.Not.Null)")]
+        public async Task WithFieldValidAssert(string assert)
+        {
+            var testCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@$"
+                #nullable enable
+                private string? s;
+                [Test]
+                public void Test()
+                {{
+                    {assert};
+                    Assert.That(s.Length, Is.GreaterThan(0));
+                }}
+
+                public void SetS(string? v) => s = v;
```

Comment:
This is needed to prevent a compile error (_warning CS0649: Field 'TestClass.s' is never assigned to, and will always have its default value null_) which causes the unit test to fail.